### ISSUE TITLE
Run Swagger UI so visitors can explore our APIs

### ISSUE THREAD

**Issue by cclauss** [internetarchive/openlibrary#5914]
Many people would like to have better, more interactive documentation on Open Library APIs.
* https://openlibrary.org/developers/api
*  https://github.com/internetarchive/openlibrary/wiki/Endpoints

### Describe the problem that you'd like solved

Run https://github.com/swagger-api/swagger-ui on an Open Library server so that visitors can better understand and try out our APIs.  #5898 could generate the required `openapi.json` file so our [Swagger UI](https://swagger.io/tools/swagger-ui) was kept up-to-date as our APIs change.  The `try it out` feature can make our APIs come alive even for nontechnical visitors.

* [x] Read https://swagger.io/tools/swagger-ui/ and https://github.com/swagger-api/swagger-ui
* [x] Recommend how we could/should run `swagger-ui` on OL infrastructure.
* [x] Get `swagger-ui` running using the [`openapi.json`](https://github.com/internetarchive/openlibrary/pull/5898/files#diff-c8dc4c1e479f3120bf2834ed4be53988bc795a39ac548200295461b6837d78cb) to drive the user experience.
    * --> https://github.com/internetarchive/openlibrary-api
    * --> https://internetarchive.github.io/openlibrary-api
* [ ] Demo on a weekly community call.
* [ ] List the [Open Library APIs](https://openlibrary.org/developers/api) that are still missing.
* [ ] Update our [Open Library API docs](https://openlibrary.org/developers/api) to be in sync with our `swagger-ui` server.
* [ ] Recommend what maintenance will be required of this new server.
    * Q: How do we update to a new version of `swagger-ui` ? -- This is done automatically by a GItHub Action.
    * Q: How do we update to a new version of [`swagger.yaml`](https://github.com/internetarchive/openlibrary-api/blob/main/swagger.yaml)?
    * Q: How can we do analytics of how many unique visitors come to the site (per hour/day/week/month/year)?

![Screenshot 2021-11-22 at 16 39 59](https://user-images.githubusercontent.com/3709715/142890924-2be77a61-c389-4385-a20f-9b543765d00c.png)

**Comment by cclauss** [comment]
@RayBB @jimman2003 Would one of you be willing to lead this effort?

**Comment by RayBB** [comment]
@cclauss what exactly needs to be checked here?

**Comment by Dnouv** [comment]
@cclauss I would like to contribute to this issue, please let me know if it's still available or not?

**Comment by Dnouv** [comment]
Thank you! I will go through the FastAPI docs.

**Comment by Dnouv** [comment]
@cclauss, I'm trying to access the API page on the development version, but it returns 404, so where and how can I access the API page on the dev version or, more specifically, the route of `/developer/API page in the repo? Could you please help me out with this?
Thank you!

**Comment by cclauss** [comment]
https://openlibrary.org/developers -- log in and click the `Edit` button in the upper right of the page.
https://openlibrary.org/developers/api  -- log in and click the `Edit` button in the upper right of the page.

**Comment by Dnouv** [comment]
Thank you so much for clearing my questions. So, if I got it correct, there is no template defined for the `/developers/api` page, and we need to create one template that integrates the Swagger-UI; please correct me if I missed something.

**Comment by cclauss** [comment]
You will need to work with @jimchamp or @cdrini or others on that because I know Python but not all that JS/CSS stuff.

**Comment by Dnouv** [comment]
> Thank you so much, @cclauss, for clearing my questions. So, if I got it correct, there is no template defined for the `/developers/api` page, and we need to create one template that integrates the Swagger-UI. And I looked at the steps of adding/integrating the SwaggerUI in the OL, and it requires a few JS files.
> 
> ```
> swagger-ui.css
> swagger-ui-bundle.js
> swagger-ui-standalone-preset.js
> ```
> 
> So, this JS files should be added in `openlibrary/plugins/openlibrary/js` or do we prefer adding `swagger-ui` [npm package](https://www.npmjs.com/package/swagger-ui)?
> 
> Please correct me if I missed something.

@cdrini or @jimchamp, could you please help me with a few questions.
Thank you!

**Comment by Dnouv** [comment]
@cclauss cdrini and jimchamp were busy, so I went ahead and used [Unpkg](https://unpkg.com/) in place of js files or npm module, I tried integrating the Swagger UI with the `openapi.json`, provided in the task list above; here is a pic of the page,

<img width="824" alt="image" src="https://user-images.githubusercontent.com/61188295/147540638-55559822-137c-49a5-bf59-24a3880ef9ac.png">

Feedbacks are most welcomed. Thank you!

**Comment by Dnouv** [comment]
@cclauss, I'm trying to commit the changes, but there is an unknown error, could you please help me out with this
```
gitpod /workspace/openlibrary $ git commit -m "Add swagger html and openapi"
`pre-commit` not found.  Did you forget to activate your virtualenv?
```

**Comment by jimman2003** [comment]
Hey, so openlibrary uses pre-commit , a python program to run some checks over the code, please install it through the pip package installer with: pip install pre-commit

**Comment by Dnouv** [comment]
Thank you so much for the help @jimman2003, I tried installing pre-commit but there was a new error
```
gitpod /workspace/openlibrary $ pip install pre-commit
pyenv: version `3.9.4' is not installed (set by /workspace/openlibrary/.python-version)
```

**Comment by jimman2003** [comment]
let me test also through gitpod :)

**Comment by Dnouv** [comment]
@jimman2003, thank you for your help! I resolved the issue.

**Comment by jimman2003** [comment]
Out of curiosity, how @Dnouv ?

**Comment by Dnouv** [comment]
I got it fixed after a few installations,
```
pyenv install 3.9.4
##Fixed pyenv: version `3.9.4' is not installed (set by /workspace/openlibrary/.python-version)

##Then your mentioned command
pip install pre-commit
```
And the changes got committed.
Once again, thank you!

**Comment by Dnouv** [comment]
@cclauss, could you please let me know the further tasks? I've opened a PR integrating Swagger UI and am still awaiting a review.
Feel free to comment. Thank you!
